Failing test case TestLinalgXPU,test_cond_errors_and_warnings_xpu_float64

### 🐛 Describe the bug

Cases:
op_ut,third_party.torch-xpu-ops.test.xpu.test_linalg_xpu.TestLinalgXPU,test_cond_errors_and_warnings_xpu_float64

<details>
<summary>Error log</summary>

```text
Traceback (most recent call last):
  File "/usr/lib/python3.12/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/usr/lib/python3.12/unittest/case.py", line 634, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.12/unittest/case.py", line 589, in _callTestMethod
    if method() is not None:
       ^^^^^^^^
  File "/home/bbela/pytorch/torch/testing/_internal/common_utils.py", line 3444, in wrapper
    method(*args, **kwargs)
  File "/home/bbela/pytorch/torch/testing/_internal/common_device_type.py", line 439, in instantiated_test
    result = test(self, **param_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bbela/pytorch/torch/testing/_internal/common_device_type.py", line 1255, in dep_fn
    return fn(slf, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bbela/pytorch/torch/testing/_internal/common_device_type.py", line 1255, in dep_fn
    return fn(slf, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bbela/pytorch/torch/testing/_internal/common_device_type.py", line 1255, in dep_fn
    return fn(slf, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bbela/pytorch/third_party/torch-xpu-ops/test/xpu/../../../../test/test_linalg.py", line 1762, in test_cond_errors_and_warnings
    self.assertEqual(len(w), 1)
  File "/home/bbela/pytorch/torch/testing/_internal/common_utils.py", line 4441, in assertEqual
    raise error_metas.pop()[0].to_error(  # type: ignore[index]
AssertionError: Scalars are not equal!

Expected 1 but got 2.
Absolute difference: 1
Relative difference: 1.0

To execute this test, run the following from the base repo dir:
    python ../../test/test_linalg.py TestLinalgXPU.test_cond_errors_and_warnings_xpu_float64

This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
```
</details>

### Versions

<details>
<summary>Versions</summary>

PyTorch version: 2.12.0a0+gitd0d73b1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Clang version: Could not collect
CMake version: version 4.2.1
Libc version: glibc-2.39

Python version: 3.12.3 (main, Jan  8 2026, 11:30:50) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.14.0-1012-intel-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: True
XPU used to build PyTorch: 20250301
Intel GPU driver version:
* intel-opencl-icd:     25.48.36300.8-1~24.04~ppa3
* libze1:       1.26.2-1~24.04~ppa1
Intel GPU models onboard:
N/A
Intel GPU models detected:
* [0] _XpuDeviceProperties(name='Intel(R) Arc(TM) B570 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE20C, uuid=86800ce2-0000-0000-0300-000000000000, driver_version='1.14.36300+8', total_memory=9663MB, local_mem_size=128KB, max_compute_units=160, memory_clock_rate=0MHz, memory_bus_width=64-bit, gpu_eu_count=144, gpu_subslice_count=18, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           39 bits physical, 48 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  16
On-line CPU(s) list:                     0-15
Vendor ID:                               GenuineIntel
Model name:                              13th Gen Intel(R) Core(TM) i5-13400
CPU family:                              6
Model:                                   191
Thread(s) per core:                      2
Core(s) per socket:                      10
Socket(s):                               1
Stepping:                                2
CPU(s) scaling MHz:                      59%
CPU max MHz:                             4600.0000
CPU min MHz:                             800.0000
BogoMIPS:                                4992.00
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt clwb intel_pt sha_ni xsaveopt xsavec xgetbv1 xsaves split_lock_detect user_shstk avx_vnni dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req hfi vnmi umip pku ospke waitpkg gfni vaes vpclmulqdq rdpid movdiri movdir64b fsrm md_clear serialize arch_lbr ibt flush_l1d arch_capabilities
Virtualization:                          VT-x
L1d cache:                               416 KiB (10 instances)
L1i cache:                               448 KiB (10 instances)
L2 cache:                                9.5 MiB (7 instances)
L3 cache:                                20 MiB (1 instance)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-15
Vulnerability Gather data sampling:      Not affected
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Reg file data sampling:    Mitigation; Clear Register File
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                     Not affected
Vulnerability Tsx async abort:           Not affected

Versions of relevant libraries:
[pip3] intel-cmplr-lib-ur==2025.3.1
[pip3] intel-openmp==2025.3.1
[pip3] mkl-include==2025.3.0
[pip3] mkl-static==2025.3.0
[pip3] numpy==2.4.1
[pip3] onemkl-license==2025.3.0
[pip3] optree==0.18.0
[pip3] tbb==2022.3.0
[pip3] tbb-devel==2022.3.0
[pip3] tcmlib==1.4.1
[pip3] torch==2.12.0a0+gitd0d73b1
[pip3] umf==1.0.2
[conda] Could not collect

</details>

--- a/test/xpu/test_linalg_xpu.py
+++ b/test/xpu/test_linalg_xpu.py
@@ -513,7 +513,7 @@ def pinv_errors_and_warnings(self, device, dtype):
     with warnings.catch_warnings(record=True) as w:
         warnings.filterwarnings(
             "ignore",
-            message=r".*Aten Op fallback from XPU to CPU happened.*",
+            message=r".*Aten Op fallback from XPU to CPU happends.*",
             category=UserWarning,
         )
         torch.linalg.pinv(a, out=out)
@@ -594,7 +594,7 @@ def cond_errors_and_warnings(self, device, dtype):
         with warnings.catch_warnings(record=True) as w:
             warnings.filterwarnings(
                 "ignore",
-                message=r".*Aten Op fallback from XPU to CPU happened.*",
+                message=r".*Aten Op fallback from XPU to CPU happends.*",
                 category=UserWarning,
             )
             # Trigger warning

--- a/test/xpu/test_linalg_xpu.py
+++ b/test/xpu/test_linalg_xpu.py
@@ -513,7 +513,7 @@ def pinv_errors_and_warnings(self, device, dtype):
     with warnings.catch_warnings(record=True) as w:
         warnings.filterwarnings(
             "ignore",
-            message=r".*Aten Op fallback from XPU to CPU happends.*",
+            message=r".*Aten Op fallback from XPU to CPU happened.*",
             category=UserWarning,
         )
         torch.linalg.pinv(a, out=out)
@@ -594,7 +594,7 @@ def cond_errors_and_warnings(self, device, dtype):
         with warnings.catch_warnings(record=True) as w:
             warnings.filterwarnings(
                 "ignore",
-                message=r".*Aten Op fallback from XPU to CPU happends.*",
+                message=r".*Aten Op fallback from XPU to CPU happened.*",
                 category=UserWarning,
             )
             # Trigger warning

--- a/test/xpu/test_linalg_xpu.py
+++ b/test/xpu/test_linalg_xpu.py
@@ -18,6 +18,7 @@ import math
 import unittest
 import warnings
 from functools import partial
+from math import inf
 
 import torch
 from torch.testing import make_tensor
@@ -565,6 +566,84 @@ def pinv_errors_and_warnings(self, device, dtype):
         torch.linalg.pinv(a, rtol=rtol)
 
 
+@dtypes(*floating_and_complex_types())
+@precisionOverride({torch.float32: 1e-3})
+def cond_errors_and_warnings(self, device, dtype):
+    norm_types = [1, -1, 2, -2, inf, -inf, "fro", "nuc", None]
+
+    # cond expects the input to be at least 2-dimensional
+    a = torch.ones(3, dtype=dtype, device=device)
+    for p in norm_types:
+        with self.assertRaisesRegex(RuntimeError, r"at least 2 dimensions"):
+            torch.linalg.cond(a, p)
+
+    # for some norm types cond expects the input to be square
+    a = torch.ones(3, 2, dtype=dtype, device=device)
+    norm_types = [1, -1, inf, -inf, "fro", "nuc"]
+    for p in norm_types:
+        with self.assertRaisesRegex(
+            RuntimeError, r"must be batches of square matrices"
+        ):
+            torch.linalg.cond(a, p)
+
+    # if non-empty out tensor with wrong shape is passed a warning is given
+    a = torch.ones((2, 2), dtype=dtype, device=device)
+    for p in ["fro", 2]:
+        real_dtype = a.real.dtype if dtype.is_complex else dtype
+        out = torch.empty(a.shape, dtype=real_dtype, device=device)
+        with warnings.catch_warnings(record=True) as w:
+            warnings.filterwarnings(
+                "ignore",
+                message=r".*Aten Op fallback from XPU to CPU happends.*",
+                category=UserWarning,
+            )
+            # Trigger warning
+            torch.linalg.cond(a, p, out=out)
+            # Check warning occurs
+            self.assertEqual(len(w), 1)
+            self.assertTrue(
+                "An output with one or more elements was resized" in str(w[-1].message)
+            )
+
+    # dtypes should be safely castable
+    out = torch.empty(0, dtype=torch.int, device=device)
+    for p in ["fro", 2]:
+        with self.assertRaisesRegex(RuntimeError, "but got result with dtype Int"):
+            torch.linalg.cond(a, p, out=out)
+
+    # device should match
+    if torch.xpu.is_available():
+        wrong_device = "cpu" if self.device_type != "cpu" else "xpu"
+        out = torch.empty(0, dtype=dtype, device=wrong_device)
+        for p in ["fro", 2]:
+            with self.assertRaisesRegex(
+                RuntimeError, "tensors to be on the same device"
+            ):
+                torch.linalg.cond(a, p, out=out)
+
+    # for batched input if at least one matrix in the batch is not invertible,
+    # we can't get the result for all other (possibly) invertible matrices in the batch without an explicit for loop.
+    # this should change when at::inverse works with silent errors
+    # NumPy works fine in this case because it's possible to silence the error and get the inverse matrix results
+    # possibly filled with NANs
+    batch_dim = 3
+    a = torch.eye(3, 3, dtype=dtype, device=device)
+    a = a.reshape((1, 3, 3))
+    a = a.repeat(batch_dim, 1, 1)
+    a[1, -1, -1] = 0  # now a[1] is singular
+    for p in [1, -1, inf, -inf, "fro", "nuc"]:
+        result = torch.linalg.cond(a, p)
+        self.assertEqual(result[1], float("inf"))
+
+    # check invalid norm type
+    a = torch.ones(3, 3, dtype=dtype, device=device)
+    for p in ["wrong_norm", 5]:
+        with self.assertRaisesRegex(
+            RuntimeError, f"linalg.cond got an invalid norm type: {p}"
+        ):
+            torch.linalg.cond(a, p)
+
+
 # Skip Float8_e4m3fnuz rowwise scaled GEMM on XPU: oneDNN lacks FNUZ support.
 # Note: the primary CUDA test is already limited to ROCm (onlyCUDA + skipCUDAIfNotRocm),
 # so this XPU variant is intentionally skipped for the same unsupported dtype.
@@ -591,6 +670,7 @@ TestLinalg.test_ck_blas_library = ck_blas_library
 TestLinalg.test_addmm_relu_tunableop_rocm = addmm_relu_tunableop_rocm
 TestLinalg.test_pinv_errors_and_warnings = pinv_errors_and_warnings
 TestLinalg.test_rotating_buffer_tunableop = rotating_buffer_tunableop
+TestLinalg.test_cond_errors_and_warnings = cond_errors_and_warnings
 TestLinalg._tunableop_ctx = __tunableop_ctx
 
 TestLinalg._default_dtype_check_enabled = True

